### PR TITLE
docs/proposals: clean up parser proposal

### PR DESCRIPTION
**Pull request by squat**
Signed-off-by: Lucas Servén Marín <lserven@gmail.com>

### CODE REVIEW

File: doc/proposals/2019_generated promql parser.md [hunk 1]
```diff
@@ -5,39 +5,43 @@ menu: proposals
...
 
-#### Producing a deeper syntax tree
-
-Consider the query:
+Consider the following PromQL query:
 
     metric{label=="value", foo~="ba."}[5m] offset 3h
 
-This is how the corresponding syntax tree looks like with the current parser:
+The cooresponding syntax tree produced by the current parser looks something like the following:
```
Comment (slrtbtfs) [inline doc/proposals/2019_generated promql parser.md]: There seems to be one o too much.

File: doc/proposals/2019_generated promql parser.md [hunk 2]
```diff
@@ -49,55 +53,66 @@ However, for using the data from the parser for things like autocompletion that
...
 
-#### Easier to maintain
+#### It is easier to maintain
 
-The complex task of recognizing the PromQL syntax is done by generated code. Humans only have to deal with a comparably readable formal grammar and the code that builds up a syntax tree from already well structured input.
+The complex task of recognizing the PromQL syntax is performed by the generated code.
+Humans only have to deal with a comparably readable formal grammar and the code that builds up a syntax tree from already well structured input.
 
-#### There exists a written down formal grammar of promql
+#### It explicitly douments the formal grammar of PromQL
```
Comment (slrtbtfs) [inline doc/proposals/2019_generated promql parser.md]: s/douments/documents/

File: doc/proposals/2019_generated promql parser.md [hunk 3]
```diff
@@ -49,55 +53,66 @@ However, for using the data from the parser for things like autocompletion that
...
 
-A formal grammar provides a human and machine readable definition of what is a correct PromQL query. This would eliminate edge cases and unknown features as described above.
+A formal grammar provides a human- and machine-readable definition of what is a correct PromQL query.
+This would eliminate edge cases and unknown features such as the ones described above.
 
 #### It is possible to handle incomplete expressions
 
-Consider this input from a user:
+Consider this following incomplete PromQL input from a user:
```
Comment (slrtbtfs) [inline doc/proposals/2019_generated promql parser.md]: Something seems wrong about that sentence. Maybe s/this/the

File: doc/proposals/2019_generated promql parser.md [hunk 4]
```diff
@@ -169,33 +184,35 @@ All Syntax tree nodes satisfy the `Expr` interface which will at least implement
...
+The changes will be confined to some name changes in types and interfaces.
 
-By Rice's theorem it might be impossible be sure whether the new parser accepts exactly the same syntax as the old one.
+### Possible Regressions
 
-Less academically it is possible that there is an edge case in the current parser that the new parser doesn't handle because one knows about it. It is likely impossible to completely mitigate the risk of accidentally introducing a breaking incompatibility here.
+By Rice's theorem, it may be impossible be certain whether the new parser accepts exactly the same syntax as the old one.
 
-This is a consequence of the fact that no formal definition of PromQL exists, yet.
+Less academically, it is possible that there exist edge cases in the current parser that the new parser does not handle because one knows about it.
```
Comment (slrtbtfs) [inline doc/proposals/2019_generated promql parser.md]: s/one/no one

File: doc/proposals/2019_generated promql parser.md (hunk 1 above)
Comment (slrtbtfs) [reply doc/proposals/2019_generated promql parser.md]: Apparently the correct spelling is corresponding  https://www.dict.cc/englisch-deutsch/corresponding.html

Comment (squat) [reply doc/proposals/2019_generated promql parser.md]: :tada:

Comment (slrtbtfs) [reply doc/proposals/2019_generated promql parser.md]: Apparently the correct spelling is "corresponding" (with two 'r's) https://www.dict.cc/englisch-deutsch/corresponding.html

Comment (squat) [reply doc/proposals/2019_generated promql parser.md]: found some more typos too :p

Comment (slrtbtfs) [reply doc/proposals/2019_generated promql parser.md]: The above comment wasn't to be sent twice. Github just didn't show it on the first attempt, that's why I sent it again.

**Comment by slrtbtfs** [comment]
Looks good now. Thanks!

**Pull request merged**